src/query.rs

```diff
@@ -195,6 +195,12 @@ impl Query {
         self.client.add_option(name, value);
         self
     }
+
+    pub fn with_param(self, name: &str, value: impl Serialize) -> Result<Self, String> {
```

Comment:
One more strange thing for me here: we tell users that client-side parameters (`bind()`) is a sort of "old style" way to do parametrization. However, the binding of columns is still done exclusively with the `?fields` syntax.  What do you think about using `{fields}` syntax?

Reply:
I like `?fields` since it can easily be escaped with `??fields`

Adding more client side syntax seems like fueling the problem